scripts/linksys-image.sh

```diff
@@ -0,0 +1,69 @@
+#!/bin/sh
+# SPDX-License-Identifier: GPL-2.0-or-later
+#
+# Copyright (C) 2024 OpenWrt.org
+#
+# This script creates a tar file for the Linksys switches of the LGS3xxC/LGS3xxMPC
+# series. It contains not only the OpenWrt firmware but additional scripts that
+# are needed for the upgrade.
+#
+# ./linksys-image.py <ImageFile> <ImageFileOut>
+
+# The check script that verifies if the images matches the hardware model
+gen_imagecheck() {
+	echo '#!/bin/sh'
+	echo 'supperflag=`cat supportlist.txt | grep -i $1 | awk '"'"'{print$2}'"'"'`'
+	echo 'if [ "${supperflag}" != "" ]; then'
+	echo 'echo 0'
+	echo 'else'
+	echo 'echo 1'
+	echo 'fi'
+}
+
+# A file with a list of supported devices. Might contain multiple lines.
+gen_supportlist() {
+	echo 'LGS310C MagicNumber 0x60402060 8380_2fib_1g 3955,1000,201,3,10,1'
```

Comment:
As you've already written on the wiki, this entry is different for every device in this series. How do you see this scale to other devices?

If you pass $DEVICE_MODEL to the script, in addition to the magic number and extra info (internal name seems to correlate with the upper 6 digits of the magic number), you can put all the required info in the Device definition. Or just pass everything after "MagicNumber" as a single string.

Reply:
Fixed

Reply:
Reducing the `imageupgrade` script to the minimally required info is a good update, but by now using a lookup table based on `DEVICE_MODEL` in linksys-image.sh and belkin-header.py you now have three locations that need to be updated to add a device. I wouldn't call that part an improvement.

Please update/revert the scripts so that device-unique properties (`BELKIN_MODEL` and e.g. `LINKSYS_MAGIC`) are defined in the makefile. That way there is only one location that needs to be updated to add a new device.

Reply:
Working alone to get the Target to 6.6 and starting this discussion feels like 2 years ago. I think it is time to stop again.

Reply:
fixed